### ISSUE TITLE
AoT broken -- allowDrop

### ISSUE THREAD

**Issue by achimha** [500tech/angular2-tree-component#150]
Using 2.7.0 (also tried vsavkin's branch) on Angular 2.4.3, I get a runtime exception when using it in AoT mode.

```
<Tree [nodes]="nodes" #tree></Tree>
```

and in the component
```
nodes = [];
```

is enough to trigger a runtime error in the change detection:

![screen shot 2017-01-20 at 15 43 20](https://cloud.githubusercontent.com/assets/6816140/22153337/39ed727c-df27-11e6-9acc-4bb518461a85.png)

The method is actually `this.parentView.context.node.allowDrop`. Any ideas?

**Comment by achimha** [comment]
This seems to be the offending statement:

https://github.com/500tech/angular2-tree-component/blob/master/lib/components/tree-node-drop-slot.component.ts#L16

```
[treeAllowDrop]="allowDrop.bind(this)">
...
allowDrop(element) {
    return this.node.options.allowDrop(element, { parent: this.node, index: this.dropIndex });
  }
```
One isn't supposed to use `this` in a template I remember but not the reason for it.

**Comment by achimha** [comment]
Some more observations, the error only happens when building in production mode, `ng build --prod --aot` in `angular-cli`.

Also the error is caused by this code line in the model, not the drop slot component referenced above:

https://github.com/500tech/angular2-tree-component/blob/master/lib/models/tree-node.model.ts#L34

It goes away if I replace
```
this.allowDrop = this.allowDrop.bind(this);
```
with
```
this.allowDrop = () => false;
```

**Comment by achimha** [comment]
PR available: https://github.com/500tech/angular2-tree-component/pull/151

**Comment by adamkleingit** [comment]
Hey @achimha,  commented inside PR - please remove all changes except the unbound change in tree-node.model.ts.

Thanks!

**Comment by adamkleingit** [comment]
I made the change manually and closed the PR.
Fixed in 2.8.0. Please verify
